File: backend/SindhDormitory.Domain/Entities/StudentProfile.cs
// =============================================================================
// Domain/Entities/StudentProfile.cs
// =============================================================================
namespace SindhDormitory.Domain.Entities;

public class StudentProfile : BaseEntity
{
    public int     ProfileId       { get; set; }
    public int     StudentId       { get; set; }
    public string? PhotoUrl        { get; set; }
    public string? GuardianName    { get; set; }
    public string? GuardianPhone   { get; set; }
    public string? GuardianRelation{ get; set; }
    public string? HomeAddress     { get; set; }
    public string? City            { get; set; }
    public string? EmergencyContact{ get; set; }
    public string? BloodGroup      { get; set; }
    public string? Disabilities    { get; set; }  // Optional, for special accommodation

    // Navigation
    public Student Student { get; set; } = null!;
}

File: backend-node/src/entities/StudentProfile.ts
import {
  Entity, PrimaryGeneratedColumn, Column, OneToOne,
  JoinColumn, CreateDateColumn, UpdateDateColumn
} from 'typeorm';
import { Student } from './Student';

@Entity('StudentProfiles')
export class StudentProfile {
  @PrimaryGeneratedColumn({ name: 'ProfileId' })
  profileId!: number;

  @Column({ name: 'StudentId', type: 'int' })
  studentId!: number;

  @Column({ name: 'PhotoUrl', type: 'varchar', length: 255, nullable: true })
  photoUrl!: string | null;

  @Column({ name: 'GuardianName', type: 'varchar', length: 150, nullable: true })
  guardianName!: string | null;

  @Column({ name: 'GuardianPhone', type: 'varchar', length: 20, nullable: true })
  guardianPhone!: string | null;

  @Column({ name: 'GuardianRelation', type: 'varchar', length: 100, nullable: true })
  guardianRelation!: string | null;

  @Column({ name: 'HomeAddress', type: 'text', nullable: true })
  homeAddress!: string | null;

  @Column({ name: 'City', type: 'varchar', length: 100, nullable: true })
  city!: string | null;

  @Column({ name: 'EmergencyContact', type: 'varchar', length: 20, nullable: true })
  emergencyContact!: string | null;

  @Column({ name: 'BloodGroup', type: 'varchar', length: 5, nullable: true })
  bloodGroup!: string | null;

  @Column({ name: 'Disabilities', type: 'text', nullable: true })
  disabilities!: string | null;

  @CreateDateColumn({ name: 'CreatedAt' })
  createdAt!: Date;

  @UpdateDateColumn({ name: 'UpdatedAt' })
  updatedAt!: Date;

  @OneToOne(() => Student, (s) => s.profile)
  @JoinColumn({ name: 'StudentId' })
  student!: Student;
}